### DISCUSSION TITLE
TGA

### DISCUSSION THREAD

**Discussion by SaharRasoulpour** [firemodels/fds#12926] [Q&A]
Hello, Hope you have a good day. I have encountered some issues in my input file. I would highly appreciate your assistance with this matter. In my sample, I have Hemicellulose that breaks down into two reactions with different A and E values. When I run my sample with a heating rate of 5°C/min (which is the default value, corresponding to 9600 s), my plot for normalized mass versus temperature appears satisfactory. However, when I change the heating rate to 10°C/min (4800s), I see unreasonable fluctuation in the middle of my trend. What could be the problem?
![hemi](https://github.com/firemodels/fds/assets/143542094/ff612106-bbe9-48de-a148-fe8f8b971d1e)
[tga_sample.txt](https://github.com/firemodels/fds/files/15300249/tga_sample.txt)

**Comment by rmcdermo** [comment]
First, I don't see anything weird about your "Numerical Result".  Perhaps you uploaded the wrong image?

Second, your input file is from an old version of FDS.  Please update to latest 6.9.1 and rework the case.  It is always possible we have fixed something, which is why we release new versions.  Thanks

**Reply by SaharRasoulpour** [reply]
Thank you for your reply. Yes, it was in the old version.

**Comment by drjfloyd** [comment]
You are not using the built-in FDS TGA feature, and instead are trying to do the TGA via the input file.  You have a couple issues with your input file for this that I  see at first glance:

-DT of 0.2 s is way too large for the timestep for heat transfer into a 10 micron thick sample.
-The FDS function for the HEAT_TRANSFER_COEFFICIENT is not going to yield a high enough HTC to represent what happens in a TGA where heat transfer to the sample is via conduction from the crucible.


I suggest you search the User's Guide for TGA and follow the examples there.

**Reply by SaharRasoulpour** [reply]
Thank you for your reply. I have one more question. In a TGA sample with six components having different mass fractions, I get accurate and correct results when I run the input file for each component separately. Each individual component yields true results on its own. However, when I combine the individual values into one input file to analyze the normalized mass versus temperature for all components together, the results are not accurate. What could be the issue with such samples?(Note: In the individual files, I set the mass fraction for each component to 1. However, when analyzing all components together, I use different mass fractions for each one, such as lipid=0.227, digest=0.099, etc.)

**Comment by SaharRasoulpour** [comment]
I have one more question. In a TGA sample with six components having different mass fractions, I get accurate and correct results when I run the input file for each component separately. Each individual component yields true results on its own. However, when I combine the individual values into one input file to analyze the normalized mass versus temperature for all components together, the results are not accurate. What could be the issue with such samples?(Note: In the individual files, I set the mass fraction for each component to 1. However, when analyzing all components together, I use different mass fractions for each one, such as lipid=0.227, digest=0.099, etc.)

**Comment by mcgratta** [comment]
It is difficult to respond to hypotheticals. You can do one of two things:
1. Use Matlab or similar tool to solve ordinary differential equations describing your reaction scheme and compare to FDS.
2. Create a very simple case with two components that demonstrates that there is something wrong with the solver or your understanding of the reaction scheme. We probably do not need six components to address the issue.

**Reply by SaharRasoulpour** [reply]
Thank you for your response. I solved the equations numerically, and when I compared the results for residual mass versus temperature for each component with the FDS results (where the FDS results are for each component separately), I found an exact match. However, my numerical results for all components combined do not exactly match the overall results from FDS (I mean for all, all reactions in one input file).

**Reply by drjfloyd** [reply]
At this point we would need an input file that demonstrates the problem you are seeing where FDS in TGA mode does not match an independent calculation using the same parameters.

**Reply by SaharRasoulpour** [reply]
Please find the attached single FDS input file for all components, which does not match with all components in the numerical method.
[tga_sample.txt](https://github.com/user-attachments/files/15516488/tga_sample.txt)

Numerical solution attached here:
<img width="360" alt="image" src="https://github.com/firemodels/fds/assets/143542094/4470e313-b207-400e-be57-dcfd0be3685e">

Why do I get an exact match with the numerical results for individual reactions, but not when considering all components together?

**Reply by drjfloyd** [reply]
As I told you two weeks ago your input file is not running FDS in TGA mode, you have set a DT of 0.2 which is likely too large for such a thin material, and you are using the default heat transfer coefficient correlation which is not likely to fully capture the heating rate of a TGA.  Folow the example in the User's Guide for running a virtural TGA and compare that against your calculation.

**Reply by drjfloyd** [reply]
Check your GLUCAN. The second reaction has an E of 0.07 which divided by R * T will be effectively zero which will make the exponent in the reaction rate 1. So at t=0 your reaction rate is 120E17.

&MATL ID                       = 'GLUCAN'
      EMISSIVITY               = 1.0
      DENSITY 	               = 1540
      CONDUCTIVITY             = 4.1E-1
      SPECIFIC_HEAT            = 2.1
      N_REACTIONS              = 1
      ALLOW_SHRINKING           = F
      HEAT_OF_REACTION(1)      = -450
      A(1)                     = 9.363454383
      E(1)                     = 269000
      N_S(1)                   = 6.86
      N_T(1)                   = 0
      NU_SPEC(1,1)             = 0.9
      SPEC_ID(1,1)             = 'NITROGEN'
      NU_MATL(1,1)             = 0.1
      MATL_ID(1,1)             = 'GLUCAN_CHAR'/
      HEAT_OF_REACTION(2)      = -450
      A(2)                     = 120.56E17
      E(2)                     = 0.072899669184
      N_S(2)                   = 6.86
      N_T(2)                   = 0
      NU_SPEC(1,2)             = 0.9
      SPEC_ID(1,2)             = 'NITROGEN'
      NU_MATL(1,2)             = 0.1
      MATL_ID(1,2)             = 'GLUCAN_CHAR'/

**Reply by SaharRasoulpour** [reply]
Yes, that was my typo. The correct values I ran are A(2) = 0.072899669184 and E(2) = 221300.

**Comment by drjfloyd** [comment]
If I change to the correct A and E values and run FDS in TGA mode and then do a quick explicit time step TGA calc in Excel I get these results.


<img width="593" alt="image" src="https://github.com/firemodels/fds/assets/12799217/b9d60a01-dc6d-4fcb-905f-aa1a21b89872">

**Reply by SaharRasoulpour** [reply]
Does FDS use an explicit time method? To change data from the inverse temperature function to the explicit time method, should I modify the source file?

**Reply by drjfloyd** [reply]
The solution method for 1D heat trasnfer is discussed in detail in the Technical Reference Guide. The FDS TGA_ANALYSIS mode still uses that method.

See the User's Guide section on testing pyrolysis models for detailed instructions and examples on using the TGA_ANALYSIS mode.

**Reply by mcgratta** [reply]
A detailed description of the 1-D solid heat conduction solver is in Appendix L of the FDS Technical Reference Guide. We use the Crank-Nicolson scheme.
